src/sql/workbench/contrib/notebook/browser/notebook.component.ts

```diff
@@ -137,6 +137,16 @@ export class NotebookComponent extends AngularDisposable implements OnInit, OnDe
 	}
 
 	ngOnInit() {
+		/// For focusing the correct cell on tabbing, we hook into the focus_in and check if the active cell
+		/// is parent of the tabbed into element.
+		this._register(DOM.addDisposableListener(window, DOM.EventType.FOCUS_IN, () => {
+			const activeCellElement = this.container.nativeElement.querySelector(`.editor-group-container.active .notebook-cell.active`);
+			if (document.activeElement.className === 'inputarea monaco-mouse-cursor-text' && !DOM.isAncestor(document.activeElement, activeCellElement)) {
+				let focusedCellElement = document.activeElement.closest('.notebook-cell');
+				let focusedCell = this.cells[+focusedCellElement.id];
```

Comment:
The ID doesn't change even if new cells are added or this one is moved so I don't think we can use that